ansible_events_ui/ruleset.py

```diff
@@ -27,18 +27,31 @@
 import tempfile
 from functools import partial
 
+import aiodocker
 import ansible_runner
-from sqlalchemy import insert, select
+from sqlalchemy import insert
 from sqlalchemy.ext.asyncio import AsyncSession
 
-from .db.models import job_instance_events, playbooks
+from ansible_events_ui.managers import taskmanager
+
+from .config import load_settings
+from .db.models import activation_instance_logs, job_instance_events
 from .managers import updatemanager
 from .messages import JobEnd
 
 logger = logging.getLogger("ansible_events_ui")
 
 activated_rulesets = {}
 ansible_events = shutil.which("ansible-events")
+ssh_agent = shutil.which("ssh-agent")
+
+
+def ensure_directory(directory):
```

Comment:
This function is not needed. Can be replaced with `os.makedirs(directory, exist_ok=True)`